### PR TITLE
Fix net-ssh default options with net-ssh < 4.0.0

### PR DESCRIPTION
**Pull request by shirosaki**
net-ssh < 4.0.0 doesn't have Net::SSH.aassing_defaults.
Fallback to logger assignment.
https://github.com/net-ssh/net-ssh/blob/v3.2.0/lib/net/ssh.rb#L211

Fix for #396 
I've confirmed `rake test:units` with net-ssh 3.2.0.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ appear at the top.
 ## [Unreleased][]
 
   * Your contribution here!
+  * [#397](https://github.com/capistrano/sshkit/pull/397): Fix net-ssh default options with net-ssh < 4.0.0 - [@shirosaki](https://github.com/shirosaki)
```
Comment (mattbrictson) [inline CHANGELOG.md]: I would rephrase the changelog entry so that people know exactly what error is fixed. Something like this:

> Fix NoMethodError `assign_defaults` with net-ssh older than 4.0.0

File: lib/sshkit/backends/netssh.rb [hunk 2]
```diff
@@ -47,7 +47,16 @@ def default_options
 
         # Set default options early for ConnectionPool cache key
         def assign_defaults
-          Net::SSH.assign_defaults(@default_options)
+          if Net::SSH.respond_to?(:assign_defaults)
+            Net::SSH.assign_defaults(@default_options)
+          else
+            # net-ssh < 4.0.0 doesn't have assing_defaults
```
Comment (mattbrictson) [inline lib/sshkit/backends/netssh.rb]: Typo: `assign_defaults`

**Comment by shirosaki** [comment]
Thanks for review.
I've modified comments.

**Comment by mattbrictson** [comment]
Great ✨
Thanks for the PR!!

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,7 @@ appear at the top.
 ## [Unreleased][]
 
   * Your contribution here!
+  * [#397](https://github.com/capistrano/sshkt/pull/397): Fix NoMethodError assign_defaults with net-ssh older than 4.0.0 - [@shirosaki](https://github.com/shirosaki)
```
Comment (reedloden) [inline CHANGELOG.md]: sshkit*

Comment (leehambley) [inline CHANGELOG.md]: I'm quite sure there's no typo here, are you certain there's an issue?

Comment (mattbrictson) [inline CHANGELOG.md]: Ah, I see there is a typo `sshkt` in the URL. I'll fix it in master. Good 👀 @reedloden
